python/lib/client/dmod/client/__main__.py

```diff
@@ -353,19 +364,28 @@ def find_client_config(basenames: Optional[List[str]] = None, dirs: Optional[Lis
 
 
 def execute_dataset_command(args, client: DmodClient):
-    async_loop = get_or_create_eventloop()
-    try:
-        result = async_loop.run_until_complete(client.data_service_action(**(vars(args))))
-        print(result)
-    except ValueError as e:
-        print(str(e))
-        exit(1)
-    except NotImplementedError as e:
-        print(str(e))
-        exit(1)
-    except Exception as e:
-        print("ERROR: Encountered {} - {}".format(e.__class__.__name__, str(e)))
-        exit(1)
+    if args.action == 'domain':
+        try:
+            print({"success": True, "domain": f"{run_domain_detection(paths=args.path).to_json()}"})
+        except DmodRuntimeError as e:
+            print({"success": False, "reason": f"{e.__class__.__name__}", "message": f"{e!s}"})
+        except Exception as e:
+            print(f"ERROR - Encountered {e.__class__.__name__} detecting domain: {e!s}")
+            exit(1)
+    else:
+        async_loop = get_or_create_eventloop()
+        try:
+            result = async_loop.run_until_complete(client.data_service_action(**(vars(args))))
+            print(result)
+        except ValueError as e:
```

Comment:
Good catch, although I'd like to handle separately since these were essentially already here (and they probably are elsewhere).  I opened #571 to track.